pages/tutorials/3.0/window/events.md

```diff
@@ -7,259 +7,329 @@
-if (event.type == sf::Event::Resized)
+if (event->is<sf::Event::Resized>())
 {
     std::cout << "new width: " << event.size.width << std::endl;
     std::cout << "new height: " << event.size.height << std::endl;
 }
 ```
 
-## The LostFocus and GainedFocus events
-
-The `sf::Event::LostFocus` and `sf::Event::GainedFocus` events are triggered when the window loses/gains focus, which happens when the user switches the currently active window. When the window is out of focus, it doesn't receive keyboard events.
+## The FocusLost and FocusGained events
 
+The `sf::Event::FocusLost` and `sf::Event::FocusGained` events are triggered when the window loses/gains focus, which happens when the user switches the currently active window.
+When the window is out of focus, it doesn't receive keyboard events.
 This event can be used e.g. if you want to pause your game when the window is inactive.
 
-There's no member associated with these events in the [`sf::Event`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1Event.php "sf::Event documentation") union.
+There's no data associated with this event.
 
 ```cpp
-if (event.type == sf::Event::LostFocus)
+if (event->is<sf::Event::FocusLost>())
     myGame.pause();
 
-if (event.type == sf::Event::GainedFocus)
+if (event->is<sf::Event::FocusGained>())
     myGame.resume();
 ```
 
 ## The TextEntered event
 
-The `sf::Event::TextEntered` event is triggered when a character is typed. This must not be confused with the `KeyPressed` event: `TextEntered` interprets the user input and produces the appropriate printable character. For example, pressing '^' then 'e' on a French keyboard will produce two `KeyPressed` events, but a single `TextEntered` event containing the 'ê' character. It works with all the input methods provided by the operating system, even the most specific or complex ones.
+The `sf::Event::TextEntered` event is triggered when a character is typed.
+This must not be confused with the `KeyPressed` event.
+`TextEntered` interprets the user input and produces the appropriate printable character.
+For example, pressing '^' then 'e' on a French keyboard will produce two `KeyPressed` events, but a single `TextEntered` event containing the 'ê' character.
+It works with all the input methods provided by the operating system, even the most specific or complex ones.
 
 This event is typically used to catch user input in a text field.
 
-The member associated with this event is `event.text`, it contains the Unicode value of the entered character. You can either put it directly in a [`sf::String`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1String.php "sf::String documentation"), or cast it to a `char` after making sure that it is in the ASCII range (0 - 127).
+You can either put this data directly in a [`sf::String`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1String.php "sf::String documentation"), or cast it to a `char` after making sure that it is in the ASCII range (0 - 127).
```

Comment:
We can't just drop the most important information, i.e. how to access the data and that it's an Unicode value.

```suggestion
The member associated with this event is `textEntered.text`, it contains the Unicode value of the entered character.
You can either put this data directly in a [`sf::String`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1String.php "sf::String documentation"), or cast it to a `char` after making sure that it is in the ASCII range (0 - 127).
```